### PR TITLE
Added a grunt build process

### LINKED ISSUE #102: npm run pretest fails
There are currently a number of eslint issues

```
$ npm run pretest

> css-layout@0.0.2 pretest /Users/colineberhardt/Projects/css-layout
> ./node_modules/eslint/bin/eslint.js src


src/Layout-test-utils.js
  418:42  error  'navigator' is not defined                no-undef
  453:45  error  expectedLayout is defined but never used  no-unused-vars

src/__tests__/Layout-test.js
  2125:10  error  Trailing comma                           no-comma-dangle
  2140:10  error  Trailing comma                           no-comma-dangle
  2155:10  error  Trailing comma                           no-comma-dangle
  2198:42  error  Trailing comma                           no-comma-dangle
  2206:54  error  Trailing comma                           no-comma-dangle
  2333:18  error  Multiple spaces found before '{'         no-multi-spaces
  2334:18  error  Multiple spaces found before '{'         no-multi-spaces
  2335:18  error  Multiple spaces found before '{'         no-multi-spaces
  2336:18  error  Multiple spaces found before '{'         no-multi-spaces
  2337:18  error  Multiple spaces found before '{'         no-multi-spaces
  2338:18  error  Multiple spaces found before '{'         no-multi-spaces
  2339:18  error  Multiple spaces found before '{'         no-multi-spaces
  2340:18  error  Multiple spaces found before '{'         no-multi-spaces
  2341:18  error  Multiple spaces found before '{'         no-multi-spaces
  2342:18  error  Multiple spaces found before '{'         no-multi-spaces
  2347:62  error  Trailing comma                           no-comma-dangle
  2348:8   error  Trailing comma                           no-comma-dangle
  2350:50  error  Multiple spaces found before 'children'  no-multi-spaces
  2376:20  error  Multiple spaces found before '{'         no-multi-spaces
  2377:20  error  Multiple spaces found before '{'         no-multi-spaces
  2378:20  error  Multiple spaces found before '{'         no-multi-spaces
  2379:20  error  Multiple spaces found before '{'         no-multi-spaces
  2380:20  error  Multiple spaces found before '{'         no-multi-spaces
  2381:20  error  Multiple spaces found before '{'         no-multi-spaces
  2382:20  error  Multiple spaces found before '{'         no-multi-spaces
  2383:20  error  Multiple spaces found before '{'         no-multi-spaces
  2384:20  error  Multiple spaces found before '{'         no-multi-spaces
  2385:20  error  Multiple spaces found before '{'         no-multi-spaces
  2390:64  error  Trailing comma                           no-comma-dangle
  2391:10  error  Trailing comma                           no-comma-dangle

src/transpile.js
   23:37  error    node is defined but never used    no-unused-vars
  277:29  warning  Multiple spaces found before '+'  no-multi-spaces
  278:29  warning  Multiple spaces found before '+'  no-multi-spaces
  279:29  warning  Multiple spaces found before '+'  no-multi-spaces
  280:29  warning  Multiple spaces found before '+'  no-multi-spaces
  282:29  warning  Multiple spaces found before '+'  no-multi-spaces
  283:29  warning  Multiple spaces found before '+'  no-multi-spaces

✖ 39 problems (33 errors, 6 warnings)
```

And linting / hinting should probably be added to the build. I'll have a look at this alongside #101

### PR DESCRIPTION
**Pull request by ColinEberhardt**
Added a more complete build process

As part of #101 :

I've created a more complete build process using Grunt, it performs the following steps:

 + Lints the code via ESLint (the ESLint version was a bit out of date, and had lots of failures, I've done a quick upgrade but have had to turn off quite a few rules in order to make progress)
 + Runs the browser tests via Karma. I've added the Karma configuration directly to the gruntfile, this just tidies things up a bit. Also, I've removed the HTML test files, these are a duplication / maintenance issue from my perspective!
 + Wraps up `Layout.js` into a universal module (UMD), via a simple grunt task that performs inclusion. This is a bit messy, and doesn't scale well if you ever want to split the code up into smaller sub-modules, but is an improvement on what we currently have.
 + Uses uglify to minify the JavaScript.

The build output sits in the `dist` folder which is added to git. I'm in two minds about this, typically I'd want to have the build output distributed via npm or a GitHub release, however, this project has transpiled output that is already part of the repo, so this doesn't feel like a pollution.

Open question:
 + Should the templated / generated C / Java also be moved into dist? This might make it clearer for consumers of this project? e.g. `dist/js`, `dist/java` ...

For my next step I want to sort out the build, currently it is part JavaScript / grunt / karma and part make. The Travis build should run everything! 

Fixes #102 #103

On another note, having the output minified, I can now submit to CDNJS :-)

### CODE REVIEW

File: src/Layout.js [hunk 1]
```diff
@@ -986,21 +986,6 @@ var computeLayout = (function() {
...
-    module.exports = factory();
-  } else {
-    // Browser globals (root is window)
-    root.returnExports = factory();
-  }
-}(this, function () {
-  return computeLayout;
-}));
+if (typeof exports === 'object') {
+  module.exports = computeLayout;
```
Comment (vjeux) [inline src/Layout.js]: Nice!

**Comment by vjeux** [comment]
Sounds good, feel free to merge when you are ready :)

**Comment by ColinEberhardt** [comment]
Thanks, I don't have merge privileges, do you want to merge this in?

**Comment by vjeux** [comment]
Woops, forgot to hit save. Now should be good

**Comment by ColinEberhardt** [comment]
Thanks, I'm almost done with unifying the Java and C builds into a single grunt build, feels much better :-)

**Comment by paramaggarwal** [comment]
Thanks for this @ColinEberhardt. I remember playing with css-layout a few months back and it was difficult to get it running. These grunt tasks made it super easy to get up and running.

**Comment by ColinEberhardt** [comment]
Thanks for the feedback, glad to hear this helped :-)

On Fri, Aug 28, 2015 at 6:28 pm, facebook/css-layout
<reply@reply.github.com>
wrote:
Thanks for this @ColinEberhardt [https://github.com/ColinEberhardt] . I remember playing with css-layout a few months back and it was difficult to
get it running. These grunt tasks made it super easy to get up and running.

—
Reply to this email directly or view it on GitHub
[https://github.com/facebook/css-layout/pull/105#issuecomment-135839507] .[https://github.com/notifications/beacon/ABDBfiHqU4syuvTH1kU7VnxnsMKT-fveks5osJHOgaJpZM4FpwNq.gif]

**Pull request merged**